### PR TITLE
type parser

### PR DESCRIPTION
**Pull request by g-harel**

### CODE REVIEW

**Review by veken1199 — DISMISSED** [review]

File: src/org/thoughtcrime/securesms/util/Stereotype.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            ObjectMapper mapper = new ObjectMapper();
+            Map<String, String> m = mapper.readValue(body, Map.class);
+            type = m.get("type");
+        } catch (IOException e) {
+            return UNKNOWN;
+        }
+
+        switch(type) {
+            case REACTION:
+         // case OTHERONE:
```
Comment (shelbaz) [inline src/org/thoughtcrime/securesms/util/Stereotype.java]: placeholder for reply to message case?

File: test/unitTest/java/org/thoughtcrime/securesms/util/StereotypeTest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Test
+    public void testReturnsUnknownWhenUnkown() {
+        assertEquals(
+            Stereotype.fromBody(JSON("unkown type")),
+            Stereotype.UNKNOWN
+        );
+    }
+
+
```
Comment (shelbaz) [inline test/unitTest/java/org/thoughtcrime/securesms/util/StereotypeTest.java]: extra line

File: test/unitTest/java/org/thoughtcrime/securesms/util/StereotypeTest.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        assertEquals(
+            Stereotype.fromBody("bad json"),
+            Stereotype.UNKNOWN
+        );
+    }
+
+    @Test
+    public void testReturnsUnknownWhenUnkown() {
+        assertEquals(
+            Stereotype.fromBody(JSON("unkown type")),
```
Comment (shelbaz) [inline test/unitTest/java/org/thoughtcrime/securesms/util/StereotypeTest.java]: typo in unknown

**Comment by veken1199** [comment]
Overall I believe this change was really important since now we will be able to easily modify the condition/structure of reaction messages by changing one line of code in this class instead of changing it in many different places in the codebase

File: src/org/thoughtcrime/securesms/util/Stereotype.java (hunk 1 above)
Comment (g-harel) [reply src/org/thoughtcrime/securesms/util/Stereotype.java]: oh yeah why not @veken1199 is it `reply`?

File: src/org/thoughtcrime/securesms/util/Stereotype.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import android.icu.text.SymbolTable;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class Stereotype {
+    public static final String REACTION = "reaction";
+    public static final String COMMENT = "comment";
```
Comment (erinb90) [inline src/org/thoughtcrime/securesms/util/Stereotype.java]: slightly misaligned

File: src/org/thoughtcrime/securesms/util/Stereotype.java [hunk 5]
```diff
@@ -26,7 +26,7 @@ public static String fromBody(String body) {
 
         switch(type) {
             case REACTION:
-            case REPLY:
+            case COMMENT:
```
Comment (veken1199) [inline src/org/thoughtcrime/securesms/util/Stereotype.java]: @g-harel can you add the tests for comment type as well ?? I can send the you exact json of a comment obj if you want

**Review by veken1199 — CHANGES_REQUESTED** [review]
missing tests coverage for comment json objects

**Review by veken1199 — APPROVED** [review]

**Pull request merged**